report-utils/extract-csv.rb

```diff
@@ -0,0 +1,46 @@
+require 'json'
+require 'csv'
+
+raise Exception, 'you must provide a result folder' unless ARGV[0]
+
+result_folder_name = ARGV[0].split('/')[ARGV[0].split('/').length - 1]
+
+puts "Creating #{result_folder_name}.csv"
+
+stats_path = ARGV[0] + '/js/stats.json'
+stats = JSON.parse(File.open(stats_path).read)
+
+# the 'group' name will be something like 'group_nooptestwithout-9eb19'
+group_keys = stats['contents'].keys.select { |key| key.to_s.match(/group/) }
+group_node = stats['contents'][group_keys[0]]
+
+MAX = 'maxResponseTime'
+MEAN = 'meanResponseTime'
+STD = 'standardDeviation'
+TOTAL = 'total'
+
+# totals row is in the 'stats' node
+totals = group_node['stats']
+
+# transaction rows are in the 'contents' node
+contents = group_node['contents']
+
+catalog = contents[contents.keys[4]]['stats']
```

Comment:
why use the number, and not the same select method as above to find the right key?

Reply:
ok, I see that now.  That json is a pain to read.  you would have to dig deeper into the hash to find the difference ,which can probably be done, but isn't worth the complexity.   Maybe toss in a comment about that.